docs/cookbook/ddd_architecture/grid.md

```diff
@@ -0,0 +1,398 @@
+    }
+}
+```
+
+### Add the grid on the Book Resource
+
+And then we create the "update" operation on our BookResource.
+
+```php
+// src/BookStore/Infrastructure/Sylius/Resource/BookResource.php
+
+// ...
+use App\BookStore\Infrastructure\Sylius\Grid\BookGrid;
+use App\BookStore\Infrastructure\Symfony\Form\BookResourceType;
+use Sylius\Resource\Metadata\AsResource;
+use Sylius\Resource\Metadata\Index;
+use Sylius\Resource\Model\ResourceInterface;
+// ...
+
+#[AsResource(
+    // ...
+    formType: BookResourceType::class, // Define the form type for all your operations
+    operations: [
+        // ...
+        new Index(
+            grid: BookGrid::class, // the grid we have just created
+        ),
+    ],
+)]
+final class BookResource implements ResourceInterface
+{
+    // ...
+}
+```
+
+### Create the Author filter type
+
+First, we need to create a Symfony form type for our custom author filter.
+
+```php
+// src/BookStore/Infrastructure/Sylius/Grid/Filter/AuthorFilterType.php
+
+declare(strict_types=1);
+
+namespace App\BookStore\Infrastructure\Sylius\Grid\Filter;
+
+use App\BookStore\Application\Query\FindBooksQuery;
+use App\BookStore\Domain\Model\Book;
+use App\Shared\Application\Query\QueryBusInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class AuthorFilterType extends AbstractType
+{
+    public function __construct(
+        private readonly QueryBusInterface $queryBus,
+    ) {
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'placeholder' => 'sylius.ui.all',
+            'choices' => $this->getChoices(),
+        ]);
+    }
+
+    public function getParent(): string
+    {
+        return ChoiceType::class;
+    }
+
+    private function getChoices(): array
+    {
+        // We do not have any findAuthorsQuery
+        // Authors are stored on the Book resource
+        $models = $this->queryBus->ask(new FindBooksQuery());
+
+        $choices = [];
+
+        /** @var Book $model */
+        foreach ($models as $model) {
+            $choices[$model->author()->value] = $model->author()->value;
+        }
+
+        ksort($choices);
+
+        return $choices;
+    }
+}
+```
+
+### Create the Author filter
+
+```php
+// src/BookStore/Infrastructure/Sylius/Grid/Filter/AuthorFilter.php
+
+declare(strict_types=1);
+
+namespace App\BookStore\Infrastructure\Sylius\Grid\Filter;
+
+use Sylius\Component\Grid\Data\DataSourceInterface;
+use Sylius\Component\Grid\Filtering\ConfigurableFilterInterface;
+
+final readonly class AuthorFilter implements ConfigurableFilterInterface
+{
+    public function apply(DataSourceInterface $dataSource, string $name, $data, array $options): void
+    {
+        throw new \RuntimeException('Not implemented'); // We cannot use the DataSource generic abstraction
```

Comment:
maybe worth explaining why not?

Reply:
Hum indeed, that could be interesting to add this on the cookbook.